app/celer-g4/RootIO.cc

```diff
@@ -140,9 +140,22 @@ void HitRootIO::WriteObject(HitRootEvent* hit_event)
 
 //---------------------------------------------------------------------------//
 /*!
- * Write, and Close or Merge output.
+ * Map sensitive detectors to contiguous IDs.
  */
-void HitRootIO::Close()
+void RootIO::AddSensitiveDetector(std::string name)
+{
+    auto iter = detector_name_id_map_.find(name);
+    if (iter == detector_name_id_map_.end())
+    {
+        detector_name_id_map_.insert({name, ++detector_id_});
+    }
```

Comment:
You can (and it's good practice to) combine search and insertion:
```suggestion
    if (iter = detector_name_id_map_. insert({std::move(name), detector_id_});
        iter != detector_id_.end())
    {
        ++detector_id_;
    }
```